adapters: improve URL connector error reporting

snkas opened the pull request:
- Log the HTTP status code at info level
- Stricter handling of `Content-Range` header, returning an error if it is present for a response which is not 206 Partial Content
- Request timeout changed from default (5s) to dedicated constant (10s)

blp commented:
> Stricter handling of Content-Range header, returning an error if it is present for a response which is not 206 Partial Content

Does this fix a bug? It's hard for me to guess whether it's strict adherence to a written standard, or whether it fixes a bug observed with or likely to be observed with some server. Frankly, it would be weird for a server to respond with a Content-Range header but not actually use that content range.

snkas commented:
This PR does not fix the bug, it makes the client more in line with the HTTP RFC (more proper handling of `Content-Range`) and adds insightful line which logs the status code (hence, which makes clear the bug by just glancing at the logs).

snkas commented:
@blp The issue tracking is: https://github.com/feldera/feldera/issues/2907

I've renamed it from "Better error handling in the URL connector." to "[adapters] Queue commands interrupt more than one URL connector" and added the High Priority tag.
